### PR TITLE
Added ability to run as a daemon in Linux along with instructions

### PR DESCRIPTION
**Pull request by strottos**
Some instructions simply say TODO at this stage and should be added when we have the ability to do this. I've looked into it and it looks like we can (at least in principle) support all the options given, possibly with a bit more work to do on the Windows side of things.

### CODE REVIEW

File: cmd/agent.go [hunk 1]
```diff
@@ -71,14 +72,45 @@ func (ar AgentRunner) Run(cmd *cobra.Command, args []string) error {
...
+				ar.logger.Error("error running instance", "error", err)
+			}
+
+			time.Sleep(time.Second * 60)
+		}
+	} else {
+		err := ar.runInstance(plugins, policyBundles)
+
+		if err != nil {
+			ar.logger.Error("error running instance", "error", err)
```
Comment (chris-cmsoft) [inline cmd/agent.go:95]:
```suggestion
			ar.logger.Error("error running instance", "error", err)
			return err
```

**Review by chris-cmsoft — CHANGES_REQUESTED** [review]

File: cmd/agent.go (hunk 1 above)
Comment (strottos) [reply cmd/agent.go:95]: I thought about this one a bit and wasn't (and still am) unsure as to the best way to handle this. With this method you'll get a retry and that seems better to me in daemon mode.

Comment (strottos) [reply cmd/agent.go:97]: Fixed and added a comment above to remind us to think about retries later.

**Review by chris-cmsoft — APPROVED** [review]

File: docs/running_as_a_service.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+### Step 2: Create a systemd service file
+
+Create a systemd service file for the ConCom agent. You can use the following:
+
+```bash
+sudo tee /etc/systemd/system/concom-agent.service <<EOF
+[Unit]
+Description=Continuous Compliance (ConCom) Agent
+Documentation=https://github.com/continuouscompliance/agent
+After=network.target local-fs.target
```
Comment (harrylincoln-CS) [inline docs/running_as_a_service.md:79]:
```suggestion
Wants=network-online.target
After=network.target network-online.target
```

`network-online` for completeness?

Comment (strottos) [reply docs/running_as_a_service.md:79]: I stole this from `containerd` I think, or something like that, but sure, can't hurt. I suppose our use case is it's pointless doing this if the machine is offline and can't send results out where you could still run `containerd` on a laptop without networking a wifi connection say. I'll commit it.

**Pull request merged**